master/buildbot/util/service.py

```diff
@@ -186,6 +186,8 @@ class AsyncService(service.Service):
     def setServiceParent(self, parent):
         if self.parent is not None:
             yield self.disownServiceParent()
+        # XXX(sa2ajj): why this magic?  The twisted version of setServiceParent
+        # does not lookg this complicated (the line below)
         parent = service.IServiceCollection(parent, parent)
```

Comment:
This is the reason why I am not so found of zope.interface (probably also why it is not in the stdlib):
It feels too magical and confuse people.

If you are into learning things, a good exercice is to look at PropertyMixin, and try to understand how it really works.